WordPress/Classes/Services/Stories/WPMediaPicker+MediaPicker.swift

```diff
@@ -161,38 +165,52 @@ extension WPMediaAsset {
     func videoURLPublisher() -> AnyPublisher<URL, Error> {
         return videoAssetPublisher().tryMap { asset -> AnyPublisher<URL, Error> in
             let filename = UUID().uuidString
-            let url = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true).appendingPathComponent(filename).appendingPathExtension("mp4")
+            let url = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true).appendingPathComponent(filename)
             let urlAsset = asset as? AVURLAsset
 
             if let assetURL = urlAsset?.url {
+                let existingExtension = assetURL.pathExtension
+                let exportURL = url.appendingPathExtension(existingExtension)
                 if urlAsset?.url.scheme != "file" {
                     // Download any file which isn't local and move it to the proper location.
                     return URLSession.shared.downloadTaskPublisher(url: assetURL).tryMap { (location, _) -> URL in
                         if let location = location {
-                            try FileManager.default.moveItem(at: location, to: url)
+                            try FileManager.default.moveItem(at: location, to: exportURL)
+                            return exportURL
+                        } else {
+                            return url
                         }
-                        return url
                     }.eraseToAnyPublisher()
                 }
-                try FileManager.default.moveItem(at: assetURL, to: url)
-                return Just(url).setFailureType(to: Error.self).eraseToAnyPublisher()
+                do {
+                    try FileManager.default.copyItem(at: assetURL, to: exportURL)
```

Comment:
We must copy the item instead of moving because we don't have permission to move the file.